discretisedfield/field.py

```diff
@@ -3059,112 +3061,98 @@ def _fromovf(cls, filename):
+        mdata_list = ['valuelabels', 'valueunits']  # information not used
+
+        header = {}
+        with open(filename, 'rb') as f:
+            # >>> READ HEADER <<<
+            header['ovf_version'] = next(f).decode('utf-8').split()[-1][-3]
+            for line in f:
+                line = line.decode('utf-8')
+                if line.startswith('# Begin: Data'):
+                    mode = line.split()[3]
+                    if mode == 'Binary':
+                        nbytes = int(line.split()[4])
                     break
-
-            listdata = list(struct.iter_unpack(formatstr,
-                                               f[data_start:data_end]))
-            datalines = np.array(listdata)
-
-            if datalines[0] != checkvalue:
-                # These two lines cannot be accessed via tests. Therefore, they
-                # are excluded from coverage.
-                msg = 'Error in checksum comparison.'  # pragma: no cover
-                raise AssertionError(msg)  # pragma: no cover
-
-            datalines = datalines[1:]
-
-        p1 = (mdatadict[key] for key in ['xmin', 'ymin', 'zmin'])
-        p2 = (mdatadict[key] for key in ['xmax', 'ymax', 'zmax'])
-        cell = (mdatadict[key] for key in ['xstepsize', 'ystepsize',
-                                           'zstepsize'])
-
-        mesh = df.Mesh(region=df.Region(p1=p1, p2=p2), cell=cell)
-
-        # valuedim is not in OVF1 file and for binary data it has to be
-        # extracted here.
-        if 'valuedim' not in mdatadict.keys():
-            mdatadict['valuedim'] = len(datalines) / len(mesh)
-
-        r_tuple = (*tuple(reversed(mesh.n)), int(mdatadict['valuedim']))
-        t_tuple = (*tuple(reversed(range(3))), 3)
-
-        return cls(mesh, dim=int(mdatadict['valuedim']),
-                   value=datalines.reshape(r_tuple).transpose(t_tuple))
+                information = line.split()
+                if len(information) > 1:
+                    key = information[1][:-1]  # remove trailing colon
+                    if key in mdata_float:
+                        header[key] = float(information[-1])
+                    elif key in mdata_int:
+                        header[key] = int(information[-1])
+                    elif key in mdata_str:
+                        header[key] = information[-1]
+                    elif key in mdata_list:
+                        header[key] = information[-3:]
+
+            if header['ovf_version'] == '1':
+                # valuedim is fixed to 3 and not in the header
+                header['valuedim'] = 3
+
+            if mode == 'Binary':
+                if header['ovf_version'] == '1':
+                    endian = '>'  # ovf1 uses big-endian
+                else:
+                    endian = '<'  # ovf2 uses little-endian
+
+            # >>> MESH <<<
+            p1 = (header[key] for key in ['xmin', 'ymin', 'zmin'])
```

Comment:
If we do not specify lists in the beginning, we can do type conversions at this point.